code/modules/atmospherics/machinery/components/components_base.dm

```diff
@@ -32,6 +32,24 @@
 		component_mixture.volume = 200
 		airs[i] = component_mixture
 
+/obj/machinery/atmospherics/components/Destroy(mob/user)
+	var/turf/T = get_turf(src)
+	var/datum/gas_mixture/to_release
+	for(var/i in 1 to device_type)
+		var/datum/gas_mixture/air = airs[i]
+		if(nodes[i])
+			var/datum/gas_mixture/parents_air = parents[i].air
+			if(parents_air.volume > 0)
+				parents_air.merge(air)
+				continue
+		if(!to_release)
+			to_release = air
+			continue
+		to_release.merge(air)
+	if(to_release)
+		T.assume_air(to_release)
+	return ..()
+
```

Comment:
Destroy should not be performing any side effects, it should delete the object and nothing more.

What are you trying to accomplish with this?

Reply:
I agree. Override  and move all this code to on_deconstruct(). 

So this code is responsible for merging/moving air from this pipe to its connected neighbour's when it is destroyed but again as @MTandi has pointed out in https://github.com/tgstation/tgstation/pull/79924#issuecomment-1826164364 this causes problems